2018/proposals/GSoC 2018 proposal Under Data Retriever.md

```diff
@@ -0,0 +1,57 @@
+
+
+## Technical Details
+* We can use these ideas to make Python code more efficient：
+   * Making the code run faster and with minimal intrusion is to use a real-time compiler (JIT). In the past, we could import psyco after installation and call psyco.full(). The code speed can be significantly improved. JIT can monitor the program in real time and compile some code into machine code.
+   * We can use image processing unit to achieve code acceleration, like PyCUDA,PyOpenGL. And these are accelerated from the hardware level. If there is a powerful GPU, we can use it to calculate, thereby reducing the valuable resources of the CPU and memory.
+   * We can use Cython, Numba. These projects are dedicated to translating Python code into C, C++, and some other code which run faster.
+   * Ctypes and can help us to achieve the operation of the Python underlying object. It can be used to build compiled C objects in memory. And call the function of C in the shared library. However, ctypes are already included in Python's standard library.
+   * After that, Julia also provides us with high code running speed and good compatibility with python. 
+* Speed and memory footprint are often not been optimized simultaneously, and we always look for their balance in program development. Through these ideas above we can make code more efficient both in speed and in weight.
+
+
+## Schedule
++ ### March 28 - May 14 **Community Bonding**
+    * Learn the basic knowledge of Data Retriever by the docset to have a clear idea with how to manage data sets. 
+    * Before starting with the project, I would like to participate in issues to find out what method is feasible.
+    * Discuss the pros and cons of the solution proposed.
++ ### May 2 - June 11 **Phase 1**
+    * Learn High-performance computing method and implement them in Python.
+    * Start work on learning the structure of funcations in Data Retriever.
+    * Put CUDA or other parallel algorithm into use to accelerate the speed of sorting or querying.
++ ### June 15 - July 8 **Phase 2**
+    * Start work on modifying the funcations in packages and modules.
+    * Optimization the speed and memory footprint both on single processor and multiprocessor.
+    * Start work on determining how fast and memory intensive areas of the codebase.
++ ### July 9- August 6 **Phase 3**
+    * Start work on adding indexes to the databases for efficient querying.
+    * Optimization the speed of searching and querying the indexes.
++ ### August 7 - 14 **Final Work**
+    * Adding docstrings,  fixing bugs， cleaning code.
+    * Passing PR reviews.
+    * Submit the code before the end of GSoC.
+    * Merge the PRs.
+## Future works
```

Comment:
Include the contributions' link to the data retriever. I gave you the link that would show all your contributions before the selection period.